src/placement/pl_map_common.c

```diff
@@ -27,26 +27,30 @@ remap_add_one(d_list_t *remap_list, struct failed_shard *f_new)
 		f_new->fs_tgt_id, f_new->fs_fseq, f_new->fs_status);
 
 	/* All failed shards are sorted by fseq in ascending order */
-	d_list_for_each_prev(tmp, remap_list) {
+	d_list_for_each(tmp, remap_list) {
 		f_shard = d_list_entry(tmp, struct failed_shard, fs_list);
 		/*
 		* Since we can only rebuild one target at a time, the
 		* target fseq should be assigned uniquely, even if all
 		* the targets of the same domain failed at same time.
 		*/
-		D_DEBUG(DB_PL, "fnew: %u, fshard: %u", f_new->fs_shard_idx,
-			f_shard->fs_shard_idx);
+		D_DEBUG(DB_PL, "fnew: %u/%u, fshard: %u/%u", f_new->fs_shard_idx,
+			f_new->fs_fseq, f_shard->fs_shard_idx, f_shard->fs_fseq);
 
-		D_ASSERTF(f_new->fs_fseq != f_shard->fs_fseq,
-			  "same fseq %u!\n", f_new->fs_fseq);
+		D_ASSERTF(f_new->fs_shard_idx != f_shard->fs_shard_idx,
+			  "same shard_idx %u!\n", f_new->fs_shard_idx);
 
-		if (f_new->fs_fseq < f_shard->fs_fseq)
+		if (f_new->fs_fseq > f_shard->fs_fseq)
 			continue;
 
-		d_list_add(&f_new->fs_list, tmp);
+		if (f_new->fs_fseq == f_shard->fs_fseq &&
```

Comment:
why is it possible to get two same fseq? the different failed tgt's "target->ta_comp.co_fseq" should never equal, no? 
Ah, is it for the case that two shard co-located on the same target then with same fseq? if so maybe can add D_ASSERT(f_new->fs_tgt_id == f_shard->fs_tgt_id), and above L29's comment need a few change.

Reply:
yeah, exactly. I will fix those comments if need refreshing the patch or update it in the following patch.